room/room-compiler-processing/src/main/java/androidx/room/compiler/processing/XValueArgument.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.compiler.processing
+
+/**
+ * This wraps information about an argument in an annotation.
+ */
+interface XValueArgument {
+    /**
+     * The property name.
+     */
+    val name: String
+
+    /**
+     * The value set on the annotation property, or the default value if it was not explicitly set.
```

Comment:
Add the possible types to docs:
* Primitives (Boolean, Byte, Int, Long, Float, Double)
* String
* XEnumEntry
* XAnnotation
* XType
* List of any of the above

Hopefully didn't miss any. =D